### PR TITLE
feat(Staking): add multiple rewards tokens

### PR DESCRIPTION
**Pull request by developerfred**
**To infinity and beyond**, 
with this pull request you have the possibility to distribute not two but several tokens with a claim.

A structure has been added: **`RewardsTokens`**

```
 struct RewardsTokens {
        address token;
        uint256 totalReward; 
    }
```

token: Contract addresses for ERC20 tokens to be distributed
totalReward: Quantity to be distributed in the staking contract

Ref
https://github.com/fuseio/fuse-lp-rewards/issues/16

### CODE REVIEW

File: solidity/contracts/Staking.sol [hunk 1]
```diff
@@ -41,19 +41,26 @@ contract Staking {
...
+     * @dev Structure to store Rewards details.
+     * Contains the address of the rewards tokens and amount of rewards distribution
+     */
+    struct RewardsTokens {
+        address token;
+        uint256 totalReward; 
+    }
+
+    // A dynamically-sized array of `RewardsTokens` structs.
+    RewardsTokens[] public RewardsTokens;
```
Comment (leonprou) [inline solidity/contracts/Staking.sol:54]: The `RewardsTokens` struct and variable got the same name. by convention var names should not be in capital

File: solidity/contracts/Staking.sol [hunk 2]
```diff
@@ -229,7 +243,9 @@ contract Staking {
         uint256 interest = calculateInterest(msg.sender);
         Staker storage stakerData = interestData.stakers[msg.sender];
         stakerData.withdrawnToDate = stakerData.withdrawnToDate.add(interest);
-        require(rewardToken.transfer(msg.sender, interest), "Withdraw interest transfer failed");
+        for (uint r = 0; r < RewardsTokens.length; r++) {
+            require(RewardsTokens[r].token.transfer(msg.sender, interest), "Withdraw interest transfer failed");
```
Comment (leonprou) [inline solidity/contracts/Staking.sol:247]: This means you send the same amount of tokens without taking into account that total reward per token is differ

**Review by leonprou — CHANGES_REQUESTED** [review]
Hey @codingsh,
Had a chance to run the tests? Part of the task is fixing them and adding sufficient cover for the new multiple reward tokens flow. Also pretty concerned with `withdrawInterest` changes.

File: solidity/contracts/Staking.sol (hunk 1 above)
Comment (developerfred) [reply solidity/contracts/Staking.sol:54]: @leonprou, I will update the tests and correct the name of the structs and modify the amount of interest according to the token.

**Comment by ashfuse** [comment]
@developerfred Can you please let us know whether you have any update?

**Comment by developerfred** [comment]
@ashfuse and @leonprou, is it ok to put the stack on hardhat? 
Tests are more organized.

**Comment by leonprou** [comment]
Hey @developerfred,
I used to work with truffle, but we are open to new stuff so if you think hardhat it better we are good with trying this.

**Comment by ashfuse** [comment]
@developerfred Any updates regarding this?

**Pull request closed without merging**